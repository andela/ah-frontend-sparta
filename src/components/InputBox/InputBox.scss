@import '../../assets/scss/shared.scss';

#loginInput {
  background-color: $primary-color;
  border-style: solid;
  border-left-color: $primary-color;
  border-right-color: $primary-color;
  border-top-color: $primary-color;
  border-bottom-color: $secondary-color;
  width: 50%;
  padding: rem(10) rem(10) rem(10) 0;
  outline: transparent;
  color: $secondary-color;
  font-size: $input-fontSize;
  font-family: $roboto-font;

  &::placeholder {
    color: $secondary-color;
    font-size: $input-fontSize;
    font-family: $roboto-font;
  }
}

.inputContain {
  margin-bottom: rem(50);
}
