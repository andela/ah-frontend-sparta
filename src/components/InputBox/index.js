import React from 'react';

import './InputBox.scss';

const InputBox = ({ name, placeholder, type, onChangeInput, value }) => {
	return (
		<div className='inputContain'>
			<input
				type={type}
				value={value}
				name={name}
				placeholder={placeholder}
				className='input'
				onChange={(val) => onChangeInput(val)}
			/>
		</div>
	);
};

export default InputBox;
