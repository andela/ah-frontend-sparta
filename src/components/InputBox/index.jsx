import React from 'react';
import PropTypes from 'prop-types';

import './InputBox.scss';

const InputBox = ({
  name, placeholder, type, onChangeInput, value,
}) => (
  <div className="inputContain">
    <input
      id="loginInput"
      type={type}
      value={value}
      name={name}
      placeholder={placeholder}
      className="input"
      onChange={onChangeInput}
    />
  </div>
);

InputBox.propTypes = {
  name: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  onChangeInput: PropTypes.func,
};

InputBox.defaultProps = {
  name: '',
  placeholder: '',
  type: '',
  value: '',
  onChangeInput: () => {},
};

export default InputBox;
