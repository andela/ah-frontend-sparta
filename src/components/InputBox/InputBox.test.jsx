import React from 'react';
import renderer from 'react-test-renderer';
import InputBox from './index';

test('snapshot test', () => {
  const wrapper = renderer.create(<InputBox />);
  expect(wrapper).toMatchSnapshot();
});
