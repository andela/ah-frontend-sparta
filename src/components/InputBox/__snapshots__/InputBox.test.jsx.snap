// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="inputContain"
>
  <input
    className="input"
    name=""
    onChange={[Function]}
    placeholder=""
    type=""
    value=""
  />
</div>
`;
